// ngx-progressbar/esm2022/lib/ng-progress.module.mjs
import { NgModule } from '@angular/core';
import { NgProgressComponent } from './ng-progress.component';
import { NG_PROGRESS_CONFIG } from './ng-progress.interface';
import * as i0 from "@angular/core";
class NgProgressModule {
    static withConfig(config) {
        return {
            ngModule: NgProgressModule,
            providers: [
                { provide: NG_PROGRESS_CONFIG, useValue: config }
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: NgProgressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.2", ngImport: i0, type: NgProgressModule, imports: [NgProgressComponent], exports: [NgProgressComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: NgProgressModule, imports: [NgProgressComponent] }); }
}
export { NgProgressModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: NgProgressModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [NgProgressComponent],
                    imports: [NgProgressComponent]
                }]
        }] });
